cumulusci/core/config/project_config.py

```diff
@@ -485,6 +491,42 @@ def _check_keychain(self):
                 + "config.set_keychain(keychain) before accessing orgs"
             )
 
+    def get_repo_from_url(self, url):
+        repo_owner, repo_name = url.split("/")[3:5]
+        if repo_name.endswith(".git"):
+            repo_name = repo_name[:-4]
+        gh = self.get_github_api(repo_owner, repo_name)
+        repo = gh.repository(repo_owner, repo_name)
+
+        return (repo, repo_owner, repo_name)
```

Comment:
Probably should get the owner and name from the repo object in case we were redirected from an old URL. Also can we just return repo and let code get the name and owner from it?